socket_exception.hpp

```diff
@@ -1,15 +1,4 @@
 #include <exception>
 #include <string>
 
-class socket_exception : public std::exception
```

Comment:
Not critical, but anyway: you'd better use single naming/styling convention in whole project.
Here: your exception class named in snake_case, while socket class in CamelCase.